apps/wing-console/console/server/src/utils/throttle.ts

```diff
@@ -0,0 +1,17 @@
+export const throttle = <T, A extends any[]>(
+  function_: (...arguments_: A) => T,
+  delay: number,
+): ((...arguments_: A) => T | undefined) => {
+  let lastTime = 0;
+  let lastCall: T | undefined;
+
+  return (...arguments_: A) => {
+    const now = Date.now();
+    if (now - lastTime < delay) {
+      return lastCall;
+    }
+    lastTime = now;
+    lastCall = function_(...arguments_);
+    return lastCall;
+  };
+};
```

Comment:
This implementation doesn't include a trailing execution, so users may not see the latest logs for a while.

I'd suggest using https://github.com/sindresorhus/p-throttle.